### ISSUE TITLE
Self loops in Tree-NeighborsMatch datasets

### ISSUE THREAD

**Issue by jhonygiraldo** [tech-srl/bottleneck#8]
Hi Uri,
 
I hope you're doing well. Your paper is really nice, congrats.
 
I just have a small question. If I correctly understood your code, you're generating several trees and then you stack them to train in a batch fashion. What I don't understand is why you add self loops to all nodes. The illustrations in the paper don't show any self loops. I guess the only difference is that you'll be using the own embedding of each node when performing message passing, isn't it? Or is there something I'm missing?
 
Thank you very much for releasing the code :)

**Comment by urialon** [comment]
Hi @jhonygiraldo,
Thank you for your interest in our work and for your kind words.

Adding self-loops just performed slightly better, so we added them. 
The purpose of these experiments was to show that this simple-looking Tree-NeighborsMatch benchmark is difficult, no matter what tricks are used. So we added self-loops and experimented with many kinds of batch normalization, layer normalization, nonlinearities, residual connections, etc.
Still, no model could perform well starting from a radius of 5.

Let me know if you have any questions!
Best,
Uri
